assembly-ide/src/assembly/bin/che.sh

```diff
@@ -10,20 +10,436 @@
+
+USAGE="
+
+Usage: 
+  che [-i] [-i:tag] [-p:port] [run | start | stop]
+
+     -i,      --image        Launches Che within a Docker container using latest image
+     -i:tag,  --image:tag    Launches Che within a Docker container using specific image tag
+     -p:port, --port:port    Port that Che server will use for HTTP requests; default=8080
+     -h,      --help         Show this help
+     run                     Starts Che application server in current console
+     start                   Starts Che application server in new console
+     stop                    Stops Che application server"
+
+function usage {
+  echo "$USAGE"
+}
+
+function error_exit {
+  echo "
+$1"
+  echo "$WRONG $CHE_VARIABLES $USAGE"
+  JUMP_TO_END=true
+}
+
+# Run the finish function if exit signal initiated
+trap exit SIGHUP SIGINT SIGTERM
+
+function parse_command_line {
+
+  ### Parse command line parameters
+  USE_DOCKER=false
+  USE_DOCKER_TAG=latest
+  USE_PORT=8080
+  USE_HELP=false
+  USE_SERVER_ACTION=run
+
+  for command_line_option in "$@"
+  do
+  case $command_line_option in
+    -i|--image)
+      USE_DOCKER=true
+      shift # past argument
+    ;;
+    -i:*|--image:*)
+      USE_DOCKER=true
+      USE_DOCKER_TAG="${command_line_option#*:}"
+      shift # past argument=value
+    ;;
+    -p:*|--port:*)
+      USE_PORT="${command_line_option#*:}"
+      shift # past argument=value
+    ;;
+    -h|--help)
+      USE_HELP=true
+      usage
+      return
+      shift # past argument
+    ;;
+    start|stop|run)
+      USE_SERVER_ACTION=${command_line_option}
+      shift # past argument with no value
+    ;;
+    *)
+            # unknown option
+    ;;
+  esac
+  done
+}
+
+function determine_os {
+  # Set OS.  Mac & Windows require VirtualBox and docker-machine.
+  win=false
+  mac=false
+  linux=false
+
+  if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    # Linux
+    linux=true
+  elif [[ "$OSTYPE" == "darwin"* ]]; then
+    # Mac OSX
+    mac=true
+  elif [[ "$OSTYPE" == "cygwin" ]]; then
+    # POSIX compatibility layer and Linux environment emulation for Windows
+    win=true
+  elif [[ "$OSTYPE" == "msys" ]]; then
+    # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
+    win=true
+  elif [[ "$OSTYPE" == "win32" ]]; then
+    # I'm not sure this can happen.
+    win=true
+  elif [[ "$OSTYPE" == "freebsd"* ]]; then
+    # FreeBSD
+    linux=true
+  else
+    # Unknown.
+    echo We could not detect your operating system.
+    echo Che is unlikely to manage Docker properly.
+  fi
+}
+
+function set_environment_variables {
+  ### Set the value of derived environment variables.
+  ### Use values set by user, unless they are broken, then fix them
+  # The base directory of Che
+  if [ -z "${CHE_HOME}" ] || [ ! -f "${CHE_HOME}" ]; then
+    export CHE_HOME="$(dirname "$(cd $(dirname ${0}) && pwd -P)")"
+  fi
+
+  # Convert Tomcat environment variables to POSIX format.
+  if [[ $JAVA_HOME == *":"* ]]
```

Comment:
Please use single format of variable expansion. Sometimes you use **$JAVA_HOME** sometimes **"${CHE_HOME}"**. Same for all variables.